test/test_plonk_component.hpp

```diff
@@ -244,12 +257,26 @@ namespace nil {
             // Stretched components do not have a manifest, as they are dynamically generated.
             if constexpr (!blueprint::components::is_component_stretcher<
                                     BlueprintFieldType, ArithmetizationParams, ComponentType>::value) {
-                BOOST_ASSERT_MSG(bp.num_gates() ==
+                BOOST_ASSERT_MSG(bp.num_gates() + bp.num_lookup_gates()==
                                 component_type::get_gate_manifest(component_instance.witness_amount(), 0,
                                                                 component_static_info_args...).get_gates_amount(),
                                 "Component total gates amount does not match actual gates amount.");
             }
 
+            if(nil::blueprint::use_lookups<component_type>()){
+                // Components with lookups may use constant columns.
+                // But now all constants are placed in the first column.
+                // So we reserve the first column for non-lookup constants. 
+                // Rather universal for testing
+                // We may start from zero if component doesn't use ordinary constants.
+                std::vector<size_t> lookup_columns_indices;
+                for( std::size_t i = 1; i < ArithmetizationParams::constant_columns; i++ )  lookup_columns_indices.push_back(i);
+                desc.usable_rows_amount = zk::snark::detail::pack_lookup_tables(
+                    bp.get_reserved_tables(),
+                    bp, assignment, lookup_columns_indices,
+                    desc.usable_rows_amount
+                );
+            }
```

Comment:
We currently do not check lookup constraints in the 
`assert(blueprint::is_satisfied(bp, assignment) == expected_to_pass);`
call on line 289. We should add some checks to satisfiability_check.hpp .